instructor/function_calls.py

```diff
@@ -295,14 +434,39 @@ def parse_json(
         )
 
 
+def openai_schema_helper(cls: T) -> T:
+    origin = get_origin(cls)
+
+    if origin is list:
+        return list[openai_schema_helper(cls.__args__[0])]
+
+    if origin is Literal:
+        return cls
+
+    if issubclass(cls, (str, int, bool, float, bytes)):
```

Comment:
I think this code doesnt handle UnionTypes, for example a BaseModel with something like
```
  name: str | None
```
At this point if you call ```issubclass(cls, (str, int, bool, float, bytes))``` on. UnionType you get ```TypeError: issubclass() arg 1 must be a class```.


Our code started crashing and I bisected it to this PR. In particular if I revert openai_schema to 
```
shema = wraps(cls, updated=())(
        create_model(
            cls.__name__ if hasattr(cls, "__name__") else str(cls),
            __base__=(cls, OpenAISchema),
        )
    )
    return cast(OpenAISchema, shema) 
```
and bypass this whole new method then it works again.

FYI we use Hakiu and i haven't validated if this code works on OpenAi, however are they any tests that cover UnionTypes?

I can also see if I can come up with a fix, but those who know the entire code base better might be faster.

Reply:
So i think if you add this
```
    if isinstance(cls, UnionType):
        for arg in get_args(cls):
            return openai_schema_helper(arg)
```
on line 444 (i.e above ```if issubclass(cls, (str, int, bool, float, bytes)): ...'```) then it works.

You will also need to 
```
from typing import Annotated, Any, Optional, TypeVar, cast, get_origin, Literal, get_args
from types import UnionType
```

Reply:
Not only that, it doesn't work for types like dict[str, str] either. The current implementation really needs to be carefully reviewed and fixed. @jxnl @ivanleomk